packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -135,7 +155,31 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ChiDiscount, ERC
         // Process the callback logic.
         _beforeBlockSubmission(blocks, config);
 
+        // Do flash deposits
+        for (uint i = 0; i < flashConfig.deposits.length; i++) {
+            IExchangeV3(target).flashDeposit(
+                flashConfig.deposits[i].to,
+                flashConfig.deposits[i].token,
+                flashConfig.deposits[i].amount
+            );
+        }
+
         target.fastCallAndVerify(gasleft(), 0, decompressed);
+
+        // Do vault logic
+        for (uint i = 0; i < flashConfig.vaults.length; i++) {
+            require(flashConfig.vaults[i].to != target, "EXCHANGE_CANNOT_BE_VAULT");
+            (bool success, ) = flashConfig.vaults[i].to.call(flashConfig.vaults[i].data);
```

Comment:
Do we need to consider how to do ERC20 authorization for `valuts[i].to` inside this contract?

Reply:
Not exactly sure what you mean, but the vault itself can do what it needs to do depending on the token and what is required for the specific case. We don't have to do anything for that here.